pydeseq2/dds.py

```diff
@@ -332,48 +313,58 @@ def fit_genewise_dispersions(self):
                 batch_size=self.batch_size,
             )(
                 delayed(fit_alpha_mle)(
-                    counts=counts_nonzero[:, i],
+                    counts=self.X[:, i],
                     design_matrix=X,
-                    mu=mu_hat_[i, :],
-                    alpha_hat=rough_disps[i],
+                    mu=self.layers["_mu_hat"][:, i],
+                    alpha_hat=self.varm["_rough_dispersions"][i],
                     min_disp=self.min_disp,
                     max_disp=self.max_disp,
                 )
-                for i in range(num_genes)
+                # for i in range(num_genes)
+                for i in self.non_zero_idx
             )
         end = time.time()
         print(f"... done in {end - start:.2f} seconds.\n")
 
         dispersions_, l_bfgs_b_converged_ = zip(*res)
-        self.genewise_dispersions = pd.Series(np.NaN, index=self.counts.columns)
-        self.genewise_dispersions.update(
-            pd.Series(dispersions_, index=self.non_zero_genes).clip(
-                self.min_disp, self.max_disp
-            )
-        )
-        self._genewise_converged = pd.Series(
-            l_bfgs_b_converged_, index=self.non_zero_genes
+
+        self.varm["genewise_dispersions"] = np.full(self.n_vars, np.NaN)
+        self.varm["genewise_dispersions"][self.non_zero] = np.clip(
+            dispersions_, self.min_disp, self.max_disp
         )
 
+        self.varm["_genewise_converged"] = np.full(self.n_vars, np.NaN)
+        self.varm["_genewise_converged"][self.non_zero] = np.array(l_bfgs_b_converged_)
```

Comment:
Not related to this PR, but this `np.array(l_bfgs_b_converged_)` does a copy of the underlying list (as they are not necessarily in contiguous areas), no? This may be inefficient, as it can be as large as the number of genes. Do you think there's a way to assign it beforehand and fill this array within `Parallel` somehow?

Reply:
Done in [8f8322b](https://github.com/owkin/PyDESeq2/pull/66/commits/8f8322be117a08e9a40bb9d71671a241c88886a8)